lib/msf/core/exploit/http/client.rb

```diff
@@ -508,6 +508,31 @@ def path_from_uri(uri)
     end
   end
 
+  #
+  # Returns a hash of request opts from a URL string
+  def request_opts_from_url(url)
+    tgt = URI.parse(url)
+    opts = { 'rhost' => tgt.host, 'rport' => tgt.port, 'uri' => tgt.request_uri }
```

Comment:
Presumably this method will always expect an absolute URL.

A URL of `/` is valid input for `URI.parse`, but `tgt` will have no `scheme` or `host` attributes.

Perhaps error handling should be performed here?

```ruby
begin
  tgt = URI.parse(url)
  raise 'Invalid URL' unless tgt.scheme =~ %r{https?}
  raise 'Invalid URL' if tgt.host.to_s.eql? ''
rescue => e
  print_error "Could not parse URL: #{e}"
  return
end
````

... or something